src/context/provideNetwork.js

```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import getDisplayName from 'react-display-name'
+
+export const networkContextType = PropTypes.shape({
+  etherscanBaseUrl: PropTypes.string,
+  name: PropTypes.string,
+})
+
+const provideNetwork = Component => {
+  const GetNetwork = (props, context) => <Component {...context} {...props} />
+  GetNetwork.contextTypes = {
+    network: networkContextType,
+  }
+  GetNetwork.displayName = `GetNetwork(${getDisplayName(Component)})`
+  return GetNetwork
+}
+
+export default provideNetwork
```

Comment:
Not sure where best to put this; `src/context/` _feels_ like it might be a good place?

We could also have a `src/context/types.js` file that hodls all the context types rather than having them separately in each context provider.